db/upgrade.php

```diff
@@ -190,5 +190,20 @@ function xmldb_tool_lockstats_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2019041502, 'tool', 'lockstats');
     }
 
+    if ($oldversion < 2019042301) {
+        $sql = "DELETE FROM {tool_lockstats_locks}
+                      WHERE " . $DB->sql_like('resourcekey', ':adhoc', true, true, true) .
```

Comment:
this seems backwards, We want to:
1) keep scheduled tasks
2) remove ad hoc tasks
3) remove everything else

So basically delete where not like scheduled

Reply:
I was also expecting to see a change to the logic here:

https://github.com/catalyst/moodle-tool_lockstats/blob/master/classes/proxy_lock_factory.php#L336

Also I think it would be better to move the 'type' detection back a step and also record this in the _locks table as well as the _history table, and then just copy this data across.